src/transformStateWithClones.js

```diff
@@ -7,7 +7,34 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const states = [];
+  const temp = { ...state };
+
+  for (const obj of actions) {
+    switch (obj.type) {
+      case 'addProperties':
+        for (const i in obj.extraData) {
+          temp[i] = obj.extraData[i];
+        }
+        states.push({ ...temp });
+        break;
+
+      case 'removeProperties':
+        for (const i in obj.keysToRemove) {
+          delete temp[obj.keysToRemove[i]];
+        }
```

Comment:
```suggestion
      temp = {};
```

Reply:
but temp is a constant. is it better to make it reassignable in this case?